HousingFinanceInterimApi.Tests/V1/UseCase/LoadActionDiaryUseCaseTests.cs

```diff
@@ -0,0 +1,166 @@
+using HousingFinanceInterimApi.V1.Exceptions;
+
+namespace HousingFinanceInterimApi.Tests.V1.UseCase;
+
+public class LoadActionDiaryUseCaseTests
+{
+    private readonly Mock<IBatchLogGateway> _mockBatchLogGateway;
+    private readonly Mock<IBatchLogErrorGateway> _mockBatchLogErrorGateway;
+    private readonly Mock<IActionDiaryGateway> _mockActionDiaryGateway;
+    private readonly Mock<IGoogleFileSettingGateway> _mockGoogleFileSettingGateway;
+    private readonly Mock<IGoogleClientService> _mockGoogleClientService;
+
+    private readonly LoadActionDiaryUseCase _classUnderTest;
+
+    public LoadActionDiaryUseCaseTests()
+    {
+        _mockBatchLogGateway = new Mock<IBatchLogGateway>();
+        _mockBatchLogErrorGateway = new Mock<IBatchLogErrorGateway>();
+        _mockActionDiaryGateway = new Mock<IActionDiaryGateway>();
+        _mockGoogleFileSettingGateway = new Mock<IGoogleFileSettingGateway>();
+        _mockGoogleClientService = new Mock<IGoogleClientService>();
+
+        var waitDurationEnvVar = 30;
+        Environment.SetEnvironmentVariable("WAIT_DURATION", waitDurationEnvVar.ToString());
+
+        _classUnderTest = new LoadActionDiaryUseCase
+            (
+                _mockBatchLogGateway.Object,
+                _mockBatchLogErrorGateway.Object,
+                _mockActionDiaryGateway.Object,
+                _mockGoogleFileSettingGateway.Object,
+                _mockGoogleClientService.Object
+            );
+
+        _mockBatchLogGateway
+            .Setup(g => g.CreateAsync(It.IsAny<string>(), It.IsAny<bool>()))
+            .ReturnsAsync(RandomGen.BatchLogDomain());
+
+        _mockBatchLogGateway
+            .Setup(g => g.CreateAsync(It.IsAny<string>(), It.IsAny<bool>()))
+            .ReturnsAsync(RandomGen.BatchLogDomain());
+
+        var rentPositionFileSettings = RandomGen.CreateMany<GoogleFileSettingDomain>(quantity: 1).ToList();
+        _mockGoogleFileSettingGateway
+            .Setup(g => g.GetSettingsByLabel(It.IsAny<string>()))
+            .ReturnsAsync(rentPositionFileSettings);
+
+        _mockGoogleClientService
+            .Setup(x => x.GetFilesInDriveAsync(It.IsAny<string>(), It.IsAny<string>()))
+            .ReturnsAsync(RandomGen.CreateMany<File>(1).ToList());
+
+        _mockGoogleClientService
+            .Setup(x => x.ReadSheetToEntitiesAsync<ActionDiaryAuxDomain>(
+                It.IsAny<string>(),
+                It.IsAny<string>(),
+                It.IsAny<string>())
+            ).ReturnsAsync(RandomGen.CreateMany<ActionDiaryAuxDomain>(quantity: 1).ToList());
+    }
+
+    [Fact]
+    public async Task ReturnsStepResponseWhenAllOK()
+    {
+        // Act
+        var useCaseCall = async () => await _classUnderTest.ExecuteAsync().ConfigureAwait(false);
+
+        // Assert
+        await useCaseCall.Should().NotThrowAsync().ConfigureAwait(false);
+
+        var stepResponse = await useCaseCall().ConfigureAwait(false);
+        stepResponse.Continue.Should().BeTrue();
```

Comment:
On line 76 you can skip the delegate declaration & you will save 2 tested method calls. You can do this because line 79 is not necessary.

If the UC returns as expected on a line 82, then it won't throw as expected on line 79. And if it throws, then it won't return. Having these 2 assertions next to each other is not needed... as the return assertion tests both cases.

Reply:
Also, while testing the `stepResponse`, it would also be good to test whether the next step time matches the delay from environment variable.